application-antivirus-api/src/main/java/com/xwiki/antivirus/internal/AntivirusJob.java

```diff
@@ -264,71 +335,125 @@ private void scanAttachment(XWikiAttachment attachment, AntivirusEngine antiviru
+                    incidents.get("deleteFailed"), incidents.get("deleted"), incidents.get("scanFailed"), startDate,
+                    endDate, e);
+            } catch (QueryException | XWikiException ex) {
+                LOGGER.error(
+                    "Failed to send the infection report. Failed to log the incidents.", ex);
+            }
         }
     }
 
-    private Map<AttachmentReference, Collection<String>> getLoggingFriendlyMap(
-        Map<XWikiAttachment, Collection<String>> xwikiAttachmentMap)
+    private JsonObject createJson(String path)
     {
-        Map<AttachmentReference, Collection<String>> attachmentReferenceMap = new LinkedHashMap<>();
-        for (Map.Entry<XWikiAttachment, Collection<String>> entry : xwikiAttachmentMap.entrySet()) {
-            attachmentReferenceMap.put(entry.getKey().getReference(), entry.getValue());
+        try {
+            List<String> jsonFileContent = Files.readAllLines(Paths.get(path));
+            JsonObject scanJson = gson.fromJson(String.join("", jsonFileContent), JsonObject.class);
+            filesScanned = scanJson.getAsJsonPrimitive(SCANNED_FILES_NR_KEY).getAsInt();
+            return scanJson;
+        } catch (IOException e) {
+            JsonObject scanJson = new JsonObject();
+            scanJson.addProperty(JOB_START_TIME_KEY, System.currentTimeMillis());
+            scanJson.addProperty(SCANNED_FILES_NR_KEY, 0);
+            scanJson.addProperty(LAST_DOCUMENT_KEY, "");
+
+            persistJson(path, scanJson);
+
+            return scanJson;
         }
+    }
 
-        return attachmentReferenceMap;
+    private String getJsonPath()
+    {
+        String jsonPath = Utils.getComponent(Environment.class).getPermanentDirectory().getAbsolutePath() + PATH;
+        try {
+            Files.createDirectories(Paths.get(jsonPath));
+        } catch (IOException e) {
+            LOGGER.warn("Could not create the path [{}] for the persisted status of the scan.", jsonPath, e);
+        }
+        jsonPath += JSON_FILE_NAME;
+        return jsonPath;
     }
 
-    private void logIncidents(Map<XWikiAttachment, Collection<String>> deletedInfectedAttachments,
-        Map<XWikiAttachment, Collection<String>> deleteFailedInfectedAttachments, String engineHint)
+    private void persistJson(String jsonPath, JsonObject scanData)
     {
-        AntivirusLog antivirusLog = Utils.getComponent(AntivirusLog.class);
+        try (PrintWriter out = new PrintWriter(new FileWriter(jsonPath))) {
+            out.write(scanData.toString());
+        } catch (IOException e) {
+            LOGGER.warn("Failed to persist Antivirus Job status file.", e);
+        }
+    }
 
-        logIncidents(antivirusLog, deletedInfectedAttachments, "deleted", engineHint);
-        logIncidents(antivirusLog, deleteFailedInfectedAttachments, "deleteFailed", engineHint);
+    private Map<String, Map<AttachmentReference, Collection<String>>> getIncidents(Date startDate)
```

Comment:
The following code is a bit duplicated. Maybe you can add a new method in ``AntivirusLog`` to retrieve the incidents.